### PR TITLE
sci-libs/kissfft: new fft library

### PR DESCRIPTION
**Pull request by epsilon-0**
new dependency of vtk

Closes: https://bugs.gentoo.org/754690
Bug: https://bugs.gentoo.org/754684
Package-Manager: Portage-3.0.18, Repoman-3.0.3
Signed-off-by: Aisha Tammy <gentoo@aisha.cc>

### CODE REVIEW

**Comment by epsilon-0** [comment]
@SoapGentoo can you take a look here, the VTK bump is gonna need this.

File: sci-libs/kissfft/kissfft-131.1.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		')
+	)
+"
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
+
+src_configure() {
+	use test && python-any-r1_pkg_setup
```
Comment (SoapGentoo) [inline sci-libs/kissfft/kissfft-131.1.0.ebuild:40]: duplicated, remove

File: sci-libs/kissfft/kissfft-131.1.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		-DKISSFFT_TOOLS=$(usex tools)
+		-DKISSFFT_USE_ALLOCA=$(usex alloca)
+		-DKISSFFT_DATATYPE=$(usex cpu_flags_x86_sse simd float)
+	)
+	cmake_src_configure
+}
+
+python_check_deps() {
+	has_version -d "dev-python/numpy[${PYTHON_USEDEP}]"
+}
```
Comment (SoapGentoo) [inline sci-libs/kissfft/kissfft-131.1.0.ebuild:53]: put this before `pkg_setup`

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sci-libs/kissfft/kissfft-131.1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	EGIT_REPO_URI="https://github.com/mborgerding/kissfft"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/mborgerding/kissfft/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="BSD"
+IUSE="alloca openmp test tools cpu_flags_x86_sse"
+SLOT="0"
```
Comment (thesamesam) [inline sci-libs/kissfft/kissfft-131.1.0.ebuild:23]: Standard ebuild block structure please -- see skel.ebuild (variable order is wrong)

File: sci-libs/kissfft/kissfft-131.1.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+if [[ ${PV} == *9999 ]] ; then
+	EGIT_REPO_URI="https://github.com/mborgerding/kissfft"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/mborgerding/kissfft/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+fi
+
+LICENSE="BSD"
+IUSE="alloca openmp test tools cpu_flags_x86_sse"
```
Comment (thesamesam) [inline sci-libs/kissfft/kissfft-131.1.0.ebuild:22]: Sort IUSE

File: sci-libs/kissfft/kissfft-131.1.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+src_configure() {
+	use test && python-any-r1_pkg_setup
+	local mycmakeargs=(
+		-DKISSFFT_OPENMP=$(usex openmp)
+		-DKISSFFT_TEST=$(usex test)
+		-DKISSFFT_TOOLS=$(usex tools)
+		-DKISSFFT_USE_ALLOCA=$(usex alloca)
+		-DKISSFFT_DATATYPE=$(usex cpu_flags_x86_sse simd float)
+	)
+	cmake_src_configure
```
Comment (thesamesam) [inline sci-libs/kissfft/kissfft-131.1.0.ebuild:48]: I usually do a newline beforehand, but it's a nit

File: sci-libs/kissfft/kissfft-131.1.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+"
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+}
+
+src_configure() {
+	use test && python-any-r1_pkg_setup
+	local mycmakeargs=(
+		-DKISSFFT_OPENMP=$(usex openmp)
```
Comment (thesamesam) [inline sci-libs/kissfft/kissfft-131.1.0.ebuild:42]: We need the magic toolchain-funcs pkg_setup and pkg_pretend checks for this, right? (See examples in tree.)

**Review by thesamesam — CHANGES_REQUESTED** [review]

File: sci-libs/kissfft/kissfft-131.1.0.ebuild (hunk 1 above)
Comment (epsilon-0) [reply sci-libs/kissfft/kissfft-131.1.0.ebuild]: done

File: sci-libs/kissfft/kissfft-131.1.0.ebuild (hunk 2 above)
Comment (epsilon-0) [reply sci-libs/kissfft/kissfft-131.1.0.ebuild:54]: done

File: sci-libs/kissfft/kissfft-131.1.0.ebuild (hunk 3 above)
Comment (epsilon-0) [reply sci-libs/kissfft/kissfft-131.1.0.ebuild:22]: done

File: sci-libs/kissfft/kissfft-131.1.0.ebuild (hunk 4 above)
Comment (epsilon-0) [reply sci-libs/kissfft/kissfft-131.1.0.ebuild]: done

File: sci-libs/kissfft/kissfft-131.1.0.ebuild (hunk 5 above)
Comment (epsilon-0) [reply sci-libs/kissfft/kissfft-131.1.0.ebuild]: done

File: sci-libs/kissfft/kissfft-131.1.0.ebuild (hunk 6 above)
Comment (epsilon-0) [reply sci-libs/kissfft/kissfft-131.1.0.ebuild:46]: add tc-check-openmp

File: sci-libs/kissfft/kissfft-131.1.0.ebuild [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+	)
+"
+
+python_check_deps() {
+	has_version -d "dev-python/numpy[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	use test && python-any-r1_pkg_setup
+	use openmp && tc-check-openmp
```
Comment (SoapGentoo) [inline sci-libs/kissfft/kissfft-131.1.0.ebuild:41]:
```suggestion
	[[ ${MERGE_TYPE} != binary ]] && use openmp && tc-check-openmp
```

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: sci-libs/kissfft/kissfft-131.1.0.ebuild (hunk 7 above)
Comment (epsilon-0) [reply sci-libs/kissfft/kissfft-131.1.0.ebuild]: done

**Pull request closed without merging**